tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -352,6 +352,46 @@ def test_update_model_version(self):
             mvd5 = self.store.get_model_version(name=mv1.name, version=mv1.version)
             self.assertEqual(mvd5.current_stage, "Staging")
 
+    def test_transition_model_version_stage(self):
+        msg = (r"Model version transition cannot archive existing model versions "
+               r"because .+ is not an Active stage. Valid stages are .+")
+
+        for stage in ["None", "Archived"]:
+            with self.assertRaisesRegex(MlflowException, msg):
+                self.store.transition_model_version_stage("model", "1", "None", True)
+
+        name = "model"
+        self._rm_maker(name)
+        mv1 = self._mv_maker(name)
+        mv2 = self._mv_maker(name)
+        mv3 = self._mv_maker(name)
+
```

Comment:
can we also check somewhere that with `False` we don't archive the existing versions?

Reply:
I'll add that test